test/support/deny_list_interceptor.ex

```diff
@@ -3,11 +3,19 @@ defmodule Bamboo.DenyListInterceptor do
 
   @deny_list ["blocked@blocked.com"]
 
-  def call(email) do
-    if Enum.any?(email.to, &(elem(&1, 1) in @deny_list)) do
+  def call(%{to: recipients} = email) when is_list(recipients) do
+    if Enum.any?(recipients, &(&1 in @deny_list)) do
       Bamboo.Email.block(email)
     else
       email
     end
   end
+
+  def call(%{to: recipient} = email) when recipient in @deny_list do
+    Bamboo.Email.block(email)
+  end
+
+  def call(email) do
+    email
+  end
```

Comment:
Hi @tarzan can you walk me through this part of the change? It's not immediately clear why we made these changes other than to use multi-function heads instead of the prior solution.

Reply:
Of course. Previously this interceptor (created for test scenario's) would expect the `to:` field to be populated by a list of tuples: `{name, email}`.
The actual test scenario's just put strings containing the email(s) on the `to:` field, so I had to check for a single binary that is an email or a list of binaries.